tests/test_publish.py

```diff
@@ -473,6 +513,84 @@ def test_update_database():
     assert db1 == db2
 
 
+def test_update_database_without_media(tmpdir):
+
+    build_root = tmpdir
+    previous_version = '1.0.0'
+    version = '1.1.0'
+
+    new_table = 'new'
+    new_files = [
+        'new/001.wav',
+        'new/002.wav',
+    ]
+    alter_files = [
+        'audio/001.wav',  # same archive as 'audio/00[2,3].wav'
+        'audio/005.wav',
+    ]
+    rem_files = [
+        'new/003.wav',  # same archive as 'audio/00[1,2].wav'
+    ]
+
+    # load without media
+    db = audb.load_to(
+        build_root,
+        DB_NAME,
+        only_metadata=True,
+        version=previous_version,
+        num_workers=pytest.NUM_WORKERS,
+        verbose=False,
+    )
+    for file in db.files:
+        assert not os.path.exists(audeer.path(build_root, file))
+
+    # update and save database
+
+    # remove files
+    for file in rem_files:
+        db.drop_files(file)
+
+    # create and alter files
+    sampling_rate = 16000
+    signal = np.ones((1, sampling_rate), np.float32)
+    for file in new_files + alter_files:
+        path = audeer.path(build_root, file)
+        audeer.mkdir(os.path.dirname(path))
+        audiofile.write(path, signal, sampling_rate)
+
+    # add new table
+    db[new_table] = audformat.Table(audformat.filewise_index(new_files))
+
+    db.save(build_root)
+
+    # publish database
+    audb.publish(
+        build_root,
+        version,
+        pytest.PUBLISH_REPOSITORY,
+        previous_version=previous_version,
+        verbose=False,
+    )
+
+    # check if missing archive files were downloaded during publish
+    assert os.path.exists(audeer.path(build_root, 'audio/002.wav'))
```

Comment:
Yes, I thought about it, too. But we would need to implement this in `audbackend`. This will definitely work with `FileSystem` and I think also with `Artifactory` (at least I think I remember that I checked it once). Though probably not all backends will support updating a zip file, we can still have a fallback where we download the whole archive, apply the changes and upload again under the hood.

I created https://github.com/audeering/audbackend/issues/42

Reply:
> we can still have a fallback where we download the whole archive, apply the changes and upload again under the hood.

That was anyway what I had in mind. But good to see, that you had an even fancier idea ;)